src/cattr/gen.py

```diff
@@ -162,11 +164,18 @@ def make_dict_structure_fn(
         # For each attribute, we try resolving the type here and now.
         # If a type is manually overwritten, this function should be
         # regenerated.
-        if type is not None:
+        if converter._prefer_attrib_converters and a.converter is not None:
```

Comment:
Instead of reading this from the converter, let's make it an argument (`_cattrs_prefer_attrib_converters`), and pass it in when calling this from the `GenConverter`.